ocaml/sdk-gen/go/templates/Enum.mustache

```diff
@@ -6,6 +6,8 @@ const (
 	//{{#doc}} {{.}}{{/doc}}
 	{{name}} {{type}} = "{{value}}"
 {{/values}}
+	// Unknown Enum Value
+	{{name}}UnknownEnumValue {{name}} = "unknown_enum_value"
```

Comment:
this is a bit nit picky but why not just `Unknown`? `EnumValue` is a bit redundant 🤔

Reply:
There're Enum has value "unknow" defined like https://github.com/xapi-project/xen-api/blob/master/ocaml/idl/datamodel.ml#L6469. I discuss with @minglumlu about this , he suggest not to use "unknown" but a not exist value like "unknown_enum_value".

Reply:
I think this means that an enum might end up with both `unknown` and `unknown_enum_value`. Ideally we should check and add Unknown only if the enum does not contain it already (I believe we already do this for c# and/or java).

Reply:
I think unlike the unknown/unrecognizable Enum value,  the "unknown" is still a meaningful and valid value from application's point of view. It would be better to distinguish them so that the SDK user can handle them separately in different levels.

Reply:
I think it might add to confusion from the implementing client's perspective. Plus that this practice has been tested in C# for a long time and is known to work ok.

Reply:
In Java we use `UNRECOGNIZED`, so maybe we can stick with that?